pkg/channelmultiplexer/channelmultiplexer_test.go

```diff
@@ -0,0 +1,54 @@
+package channelmultiplexer
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type msgstruct struct {
+	i   int
+	str string
+}
+
+func TestMultiplexingChannels(t *testing.T) {
+	cases := map[string]struct {
+		inputChannels []chan *string
```

Comment:
check if this really needs to be chan *string, otherwise make it chan string

Reply:
Make 3 channels before the test and close them afterwards

Reply:
Oh, shall we then change the template to be `<-chan T` instead of `<-chan *T`?

Reply:
It looks like we forgot the "Make 3 channels before the test and close them afterwards"

Reply:
With 3 channels I got race conditions and the test got stuck. Repeated usage of a channel between test cases I think led to the multiplexer of the first test sucking up the message for the second test. It works with 5 channels.

Reply:
I refactored it a bit additionally